reporter/reporter.go

```diff
@@ -55,6 +65,7 @@ func New(cfg *config.ReporterConfig, btcClient btcclient.BTCClient, babylonClien
 		maxRetrySleepTime:             maxRetrySleepTime,
 		btcClient:                     btcClient,
 		babylonClient:                 babylonClient,
+		babylonQuerier:                q,
```

Comment:
It is actually not necessary to add a querier to reporter. The `babylonClient` here already inherits all query functions. That is, babylonQueryClient is a subset of babylonClient. See https://github.com/babylonchain/rpc-client/blob/dev/client/client.go#L14-L15 and https://github.com/babylonchain/rpc-client/blob/dev/client/client.go#L32-L36. Thus we don't need to touch the reporter in this PR.

Reply:
Thing is that it would be better to use interfaces of Babylon rpc-client other than concrete structs, but those interfaces are separated in [BabylonClient](https://github.com/babylonchain/rpc-client/blob/dev/client/interface.go) and [BabylonQueryClient](https://github.com/babylonchain/rpc-client/blob/dev/query/interface.go).

Reply:
Then maybe we can do the following?
```suggestion
		babylonQuerier:                babylonClient,
```

This way we don't need to open two concurrent RPC clients for the same node.

Also alternatively, would it be better if `BabylonClient` interface can inherit `BabylonQueryClient` as follows?

```go
type BabylonClient interface {
	BabylonQueryClient
	Stop()
	GetConfig() *config.BabylonConfig
	GetTagIdx() uint8
	GetAddr() (sdk.AccAddress, error)
	MustGetAddr() sdk.AccAddress
	InsertBTCSpvProof(msg *btcctypes.MsgInsertBTCSpvProof) (*sdk.TxResponse, error)
	InsertHeader(msg *btclctypes.MsgInsertHeader) (*sdk.TxResponse, error)
	InsertHeaders(msgs []*btclctypes.MsgInsertHeader) (*sdk.TxResponse, error)
}
```

Reply:
Yeah, I would prefer the second approach where `BabylonClient` inherits `BabylonQueryClient`

Reply:
Yeah I have the same impression. Done https://github.com/babylonchain/rpc-client/pull/30